lxd/device/nic_bridged.go

```diff
@@ -0,0 +1,1344 @@
+package device
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/big"
+	"math/rand"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/mdlayher/eui64"
+
+	"github.com/lxc/lxd/lxd/device/config"
+	"github.com/lxc/lxd/lxd/dnsmasq"
+	"github.com/lxc/lxd/lxd/instance"
+	"github.com/lxc/lxd/lxd/iptables"
+	"github.com/lxc/lxd/shared"
+)
+
+// dhcpAllocation represents an IP allocation from dnsmasq used for IP filtering.
+type dhcpAllocation struct {
+	IP     net.IP
+	Name   string
+	MAC    net.HardwareAddr
+	Static bool
+}
+
+// dhcpRange represents a range of IPs from start to end.
+type dhcpRange struct {
+	Start net.IP
+	End   net.IP
+}
+
+type nicBridged struct {
+	deviceCommon
+}
+
+// valite checks the supplied config for correctness.
+func (d *nicBridged) validate() error {
+	if d.instance.Type() != instance.TypeContainer {
+		return ErrUnsupportedDevType
+	}
+
+	if d.config["parent"] == "" {
+		return fmt.Errorf("Requires parent property")
+	}
+
+	managedNetworkFields := []string{"ipv4.address", "ipv6.address", "security.mac_filtering", "security.ipv4_filtering", "security.ipv6_filtering"}
+
+	// If parent isn't a managed network, check no managed-only features aren't enabled.
```

Comment:
The double negative is confusing me :)
Shouldn't that just be `are enabled.`?

Reply:
@stgraber indeed, it confuses me too, fixed. :)